test/server/graphql/v2/query/HostMetricsQuery.test.js

```diff
@@ -99,6 +116,17 @@ describe('server/graphql/v2/query/HostMetricsQuery', () => {
         expect(hostMetrics.hostFees.valueInCents).to.equal(300);
         expect(hostMetrics.hostFeeShare.valueInCents).to.equal(45);
       });
+
+      it('correctly calculates totalMoneyManaged', async () => {
+        const dateFrom = new Date('2021-01-01').toISOString();
+        const dateTo = new Date().toISOString();
+        const variables = { slug: host.slug, dateFrom, dateTo };
+        const queryResponse = await graphqlQueryV2(hostMetricsQuery, variables);
+        const hostMetrics = queryResponse.data.host.hostMetrics;
+        const hostMetricsTimeSeriesNodes = queryResponse.data.host.hostMetricsTimeSeries.totalMoneyManaged.nodes;
+        const totalMoneyManaged = sumBy(hostMetricsTimeSeriesNodes, node => node.amount.valueInCents);
+        expect(hostMetrics.totalMoneyManaged.valueInCents).to.equal(totalMoneyManaged);
```

Comment:
This one I am not exactly sure tbh. 🤔 My understanding is that the `hostMetrics.totalMoneyManaged` gives the cumulative amount of the money managed by the host. For example see,

https://github.com/opencollective/opencollective-api/blob/e0a67eb968f94357da3b1257e6d5c4d5879fd1d5/server/lib/budget.js#L192-L198

where the total money managed calculation is done. This seems to be a sum of all the transactions. This is why I thought that those two should match. But I understand your point where the money managed should be how much money is managed currently by the host (for the given month). But in this case I think we need to probably should change the heading; `Total Money Managed` or indicate somewhere that it really reflects the amount of money managed by the host for the given month? 🤔

Reply:
Yes, to get the amount managed for a given date you'll need the query to do something like:

```
SUM(credits) - SUM(debits)
WHERE "createdAt" <= "myDate"
```

Your approach was indeed not crazy in that sense, but since you were filtering on `createdAt` transactions created before `startDate` were not included either so we always considered that you start from $0 on `startDate`. @znarf can maybe help you to build this one.

Reply:
With a simplistic approach, sounds like a somewhat intensive thing to compute total money managed over time (iterating on dates on computing the balance).

Reply:
@Betree @znarf : For this one, I've adjusted the query and the tests so that we only consider the following transaction kinds; 

https://github.com/opencollective/opencollective-api/blob/902d7b7914e18dc37891a10628f00881e80d0a84/server/lib/host-metrics.js#L269

My thinking is we should not consider things like platform tips and payment processor fee here. Let me know hoe it looks. 

@Betree : So for the tests I've adjusted them such that I always pass the `startDate` as the `host.createDate` so that we have this type of query which makes sense. 

```
SUM(credits) - SUM(debits)
WHERE "createdAt" <= "myDate"
```

However I feel that this might create confusion for users in the front-end. The users have a period filter and ideally to get the stats right we have to ignore the `startDate` and only consider the `endDate`. Do you think we need some other design change on the front-end so that users know exactly that in the money managed section we don't consider the `startDate`? 🤔